text/0000-build-std.md

```diff
@@ -0,0 +1,4582 @@
+- [*Why not use weak linkage for `compiler-builtins/mem` symbols?*][rationale-no-weak-linkage]
+
+### `libunwind`
+[libunwind]: #libunwind
+
+`libunwind`'s sources are included in the `rust-src` component so that they can
+be used as part of the standard library build on targets which require it.
+
+### Potential migration breakage
+[potential-migration-breakage]: #potential-migration-breakage
+
+When building an existing `no_std` project for a tier two or three target with
+build-std, there could be an implicit dependency on `std` from a dependency
+`no_std` crate that has not yet made its dependency on only the `core` crate
+explicit, for example. In this circumstance, this would fail to build as the
+target will have had `standard_library_support.std = false` in its target
+specification and Cargo will refuse to build `std` (see
+[*Target standard library support*][target-standard-library-support])
+([?][rationale-breakage]).
+
+*See the following sections for rationale/alternatives:*
+
+- [*Why permit breakage of nightly build-std users using tier three targets?*][rationale-breakage]
+
+### Caching
+[caching]: #caching
+
+Standard library artifacts built by build-std will not be shared between crates
+or workspaces, as they only exist in the `target` directory of a specific crate
+or workspace ([?][rationale-caching]).
+
+*See the following sections for rationale/alternatives:*
+
+- [*Why not globally cache builds of the standard library?*][rationale-caching]
+
+### Sanitizers
+[sanitizers]: #sanitizers
+
+rustc's sanitizer support is currently unstable as it is not possible for users
+to re-build the standard library with sanitizer support.
+
+It is out-of-scope for this RFC to propose stabilising sanitizers (see
+[rust#123617]) or to expose sanitizer configuration in Cargo, but it is
+instructive to examine how build-std would enable sanitizer support to ensure
+that the proposed design is compatible.
+
+rustc's flag to enable sanitizers will be a target modifier, as the
+instrumentation must be present for all of the crates to avoid false negatives.
+
+rustc's sanitizer support attempts to locate sanitizer runtimes in the sysroot
+(`$sysroot/lib/rustlib/$target/lib/`) to link against. Rust already ships
+sanitizer runtimes for targets that support sanitizers and with the use of
+`build-std` as proposed in this RFC all Rust crates in a crate graph should have
+coverage with the requested sanitizers.
+
+Combining these shipped sanitizer runtimes with other target modifiers is
+outside the scope of this RFC.
+
+## Cargo subcommands
+[cargo-subcommands]: #cargo-subcommands
+
+As opaque dependencies, any Cargo command which accepts a package spec with `-p`
+will only additionally recognise `core`, `alloc` and `std` and none of their
+dependencies. Many of Cargo's subcommands will need modification to support
+build-std:
+
+[`cargo add`][cargo-add] will add `core`, `alloc` or `std` explicitly to the
+manifest if invoked with those crate names:
```

Comment:
Hmm, interesting. Perhaps it should also enforce the nightly toolchain when trying to add it with features.